proxy_agent/src/proxy/authorization_rules.rs

```diff
@@ -37,107 +29,198 @@ pub struct AuthorizationRules {
+        match dict {
+            Some(dict) => dict.contains_key(&key),
+            None => false,
+        }
+    }
+
+    fn get_from_dict<T>(dict: &Option<HashMap<String, T>>, key: String) -> Option<T>
+    where
+        T: Clone,
+    {
+        match dict {
+            Some(dict) => dict.get(&key).cloned(),
+            None => None,
         }
     }
 
     pub fn is_allowed(&self, connection_id: u128, request_url: hyper::Uri, claims: Claims) -> bool {
         if self.mode.to_lowercase() == "disabled" {
+            Connection::write_information(
+                connection_id,
+                "Access control is in disabled state, skip....".to_string(),
+            );
+
             return true;
         }
 
-        if let Some(rules) = &self.rules {
-            let mut role_privilege_matched = false;
-            for rule in rules {
-                // is privilege match
-                for privilege in &rule.privileges {
-                    if privilege.is_match(connection_id, request_url.clone()) {
-                        role_privilege_matched = true;
-                        for identity in &rule.identities {
-                            if identity.is_match(connection_id, claims.clone()) {
-                                return true;
+        if let (Some(privileges), Some(identities), Some(privilege_assignments)) = (
+            &self.privileges,
+            &self.identities,
+            &self.privilegeAssignments,
+        ) {
+            let mut privilege_matched = false;
+            for privilege in privileges.values() {
+                let privilege_name = privilege.name.clone();
+                if privilege.is_match(connection_id, request_url.clone()) {
+                    privilege_matched = true;
+                    Connection::write_information(
+                        connection_id,
+                        format!("Reqeust matched privilege '{}'.", privilege_name),
+                    );
+
+                    if let Some(assignments) = privilege_assignments.get(&privilege_name) {
+                        for assignemnt in assignments {
+                            let identity_name = assignemnt.clone();
+                            if let Some(identity) = identities.get(&identity_name) {
+                                if identity.is_match(connection_id, claims.clone()) {
+                                    Connection::write_information(
+                                        connection_id,
+                                        format!("Identity matched identity '{}'.", identity_name),
+                                    );
+                                    return true;
+                                } else {
+                                    Connection::write_information(
+                                        connection_id,
+                                        format!(
+                                            "Identity does not match identity '{}'.",
```

Comment:
It's too verbose to log each branch of each check, especially with full sentences. I think a summary at the end will be more readable for users.

It could cover the highlights like if default was used, which privilege matched if one matched. If none matched, we don't need each none printed it's implied by that high level summary

Reply:
removed.